packages/go_router/lib/src/delegate.dart

```diff
@@ -204,6 +184,84 @@ class GoRouterDelegate extends RouterDelegate<RouteMatchList>
   }
 }
 
+/// An iterator that iterates through navigators that [GoRouterDelegate]
+/// created from the inner to outer.
+///
+/// The iterator starts with the navigator that hosts the top-most route. This
+/// navigator may not be the inner-most navigator if the top-most route is a
+/// pageless route, such as a dialog or bottom sheet.
```

Comment:
Why not? Isn't it okay to have an inner Navigator with a page-backed route and a pageless route?

Reply:
Am I understanding correctly that this function returns a `Iterator<NavigatorState>` in the order that `maybePop()` should be called?

Reply:
> Why not? Isn't it okay to have an inner Navigator with a page-backed route and a pageless route?

Consider this case

matchlist: [ShellRoute, GoRouteA]

user in GoRouteA and called showDialog to push a dialog to root navigator. In that case the maybepop should be call on the root navigator instead of the shellroute navigator.

> Am I understanding correctly that this function returns a Iterator<NavigatorState> in the order that maybePop() should be called?

correct.